[WhiteBear82] iP

WhiteBear82 opened the pull request:
# WinterBearBot 🐻❄️
> “Winter is the time for comfort, for good food and warmth, for the touch of a friendly hand and for a talk beside the fire.” – Edith Sitwell [(source)](https://www.goodreads.com/quotes/124075-winter-is-the-time-for-comfort-for-good-food-and)

WinterBearBot is your cozy assistant for managing tasks, deadlines, and events. It's designed to be:
* User-friendly
* 🧊 Cool to use
* ⏱️ Efficient

Getting started is simple:
1. Clone the repository from [here](https://github.com/WhiteBear82/ip/releases).
2. Run the program and start adding your tasks.
3. Manage your tasks like a pro! 😎

And it’s **completely free!** 🎉

Features:

- [X] Add tasks, deadlines, or events effortlessly.
- [ ] Manage your to-do list with actions like marking as done, unmarking, deleting
- [ ] List all tasks in a clean, organized manner.

```java
public class WinterBearBot {
    public static void main(String[] args) {
        new WinterBearBot().runProgram();
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

WhiteBear82 committed:
test commit

WhiteBear82 committed:
initial commit

WhiteBear82 committed:
Completed Level-0, added WinterBearBot.java

WhiteBear82 committed:
Completed Level-1, updated WinterBearBot.java

WhiteBear82 committed:
Completed Level-2, updated WinterBearBot.java

WhiteBear82 committed:
Completed Level-3, updated WinterBearBot.java, added Task.java

WhiteBear82 committed:
Completed Level-4, updated WinterBearBot.java, Task.java, added Todo.java, Deadline.java, Events.java

WhiteBear82 committed:
Completed Level-5, added test files into text-ui-test and tested the WinterBearBot.

WhiteBear82 committed:
Completed Level-5, added test files into text-ui-test, created enum TaskType.java, exception class WBBException.java, updated Todo.java, WinterBearBot.java, and tested the WinterBearBot.

WhiteBear82 committed:
Updated Level-5, added test files into text-ui-test, created enum TaskType.java, exception class WBBException.java, updated Todo.java, WinterBearBot.java, and tested the WinterBearBot.

WhiteBear82 committed:
Updated Level-5, added test files into text-ui-test, created enum TaskType.java, exception class WBBException.java, updated Todo.java, WinterBearBot.java, and tested the WinterBearBot.

WhiteBear82 committed:
Updated Level-6, updated WinterBearBot.java and test-ui-test files, and tested the WinterBearBot.java

WhiteBear82 committed:
Updated Level-6, updated WinterBearBot.java and test-ui-test files, and tested the WinterBearBot.java

WhiteBear82 committed:
Created A-Enums, enum TaskType.java, updated WinterBearBot.java, and tested the WinterBearBot.java

WhiteBear82 committed:
Complete Level-7, Add Storage.java for saving/loading Tasks, Add methods in Enum TaskType.java, Edit WinterBearBot.java to implement save/load tasks

WhiteBear82 committed:
Merge branch-Level-7 into master

WhiteBear82 committed:
Complete Level-8, Modify Deadline.java, Task.java, WinterBearBot.java to recognise datetime and date

WhiteBear82 committed:
Merge branch-Level-8

WhiteBear82 committed:
Create additional files for a more OOP concept.

WhiteBear82 committed:
Merge branch-A-MoreOOP

WhiteBear82 committed:
Created folders under src main java and divide files into packages

WhiteBear82 committed:
Merge branch-A-Packages

WhiteBear82 committed:
Merge branch 'add-gradle-support'

WhiteBear82 committed:
Add and Passed JUnit Tests

WhiteBear82 committed:
Merge branch 'branch-A-JUnit'

WhiteBear82 committed:
Add WinterBearBot.jar

WhiteBear82 committed:
merge branch-A-Jar

WhiteBear82 committed:
Update A-JavaDoc

WhiteBear82 committed:
Merge branch 'branch-A-JavaDoc'

WhiteBear82 committed:
Comply coding standards

WhiteBear82 committed:
Merge branch 'branch-A-CodingStandard'

WhiteBear82 committed:
Add FindCommand.java and JUnit test cases for FindCommand.java

WhiteBear82 committed:
Merge branch-Level-9

WhiteBear82 committed:
Updated A-JUnit

WhiteBear82 committed:
Ensure bot sends welcome message at startup

WhiteBear82 committed:
Ensure bot sends welcome message at startup

WhiteBear82 committed:
Fix bot response for invalid first commands

WhiteBear82 committed:
Merge branch 'branch-Level-10'

WhiteBear82 committed:
Resolve NullPointerException in getResponse

WhiteBear82 committed:
Resolve NullPointerException in getResponse

WhiteBear82 committed:
Fix scroll pane auto-scrolling issue in UI

WhiteBear82 committed:
Fix scroll pane auto-scrolling issue in UI

WhiteBear82 committed:
Add support for displaying error messages in dialog box

WhiteBear82 committed:
Add support for displaying error messages in dialog box

WhiteBear82 committed:
Introduce user and bot avatars for dialog messages

WhiteBear82 committed:
Introduce user and bot avatars for dialog messages

WhiteBear82 committed:
Improve welcome message formatting for better clarity

WhiteBear82 committed:
Improve welcome message formatting for better clarity

WhiteBear82 committed:
Add icons for user and bot messages in chat

WhiteBear82 committed:
Add icons for user and bot messages in chat

WeiHungLoh commented on src/main/java/wbb/command/Command.java:
```diff
@@ -0,0 +1,32 @@
+import wbb.storage.Storage;
+import wbb.exception.WBBException;
+
+public abstract class Command {
```
I like how you designed an abstract class for commands and extended it with different command types.

WeiHungLoh commented on src/main/java/wbb/command/DeleteCommand.java:
```diff
@@ -0,0 +1,38 @@
+     * @param ui The ui.
+     * @param storage The storage.
+     * @throws WBBException if the command cannot be executed correctly.
+     */
```
I noticed you missed out @Override annotations for the execute method in some command classes.

WeiHungLoh commented on src/main/java/wbb/command/ChangeStatusCommand.java:
```diff
@@ -0,0 +1,43 @@
+public class ChangeStatusCommand extends Command {
+    /**
+     * Executes a given command.
+     * @param taskList The taskList.
```
It's good that you have included javadoc comments for your classes and methods. However, there should be a space between the method description and the @param section. In this case, a space is needed between "Executed a given command." and "@param tasklist ..."

If adding space causes "./gradlew checkstyleMain checkstyleTest" to fail, you might need to navigate to the affected line and adjust the spacing by pressing backspace once without removing the extra line with "*".

WeiHungLoh commented on src/main/java/wbb/exception/WBBException.java:
```diff
@@ -0,0 +1,14 @@
+/**
+ * WinterBearBot Exception Class.
+ */
+public class WBBException extends Exception {
```
While not required, perhaps you may consider adding classes like InvalidCommandException or EmptyInputException that extend WBBException for clarity/debugging purposes?

WeiHungLoh commented on src/main/java/wbb/util/Validator.java:
```diff
@@ -0,0 +1,136 @@
+
+public class Validator {
+    /**
+     * To retrieve, validate and return the item (task) index.
```
It's good that you included detailed Javadoc comments for most of your methods and classes. However, perhaps you may consider changing the method header comment to "Retrieves ..." instead of "To retrieve, ..." to align with Javadoc conventions.

WeiHungLoh reviewed (commented):
Looks good to me overall. Keep up the good work!

WhiteBear82 committed:
Implement CheckStyle to enforce coding standards in Java

WhiteBear82 committed:
Implement CheckStyle to enforce coding standards in Java

WhiteBear82 committed:
Use varargs to handle command inputs

WhiteBear82 committed:
Use varargs to handle command inputs

WhiteBear82 committed:
Add assertions for robustness

Assertions are missing in both the Storage and Task classes, which
leads to undetected errors when assumptions about data structures or
inputs are violated.

Without assertions, unexpected null values, invalid task formats, and
other inconsistencies do not get caught early, making debugging harder.

Let's add assertions to enforce key assumptions, such as:
* Ensuring non-null task lists before saving.
* Validating task format and type before parsing.
* Checking that files and directories are properly initialized.

These assertions catch issues earlier during development and improve
code reliability.

WhiteBear82 committed:
Add assertions for robustness

Assertions are missing in both the Storage and Task classes, which
leads to undetected errors when assumptions about data structures or
inputs are violated.

Without assertions, unexpected null values, invalid task formats, and
other inconsistencies do not get caught early, making debugging harder.

Let's assertions to enforce key assumptions, such as:
* Ensuring non-null task lists before saving.
* Validating task format and type before parsing.
* Checking that files and directories are properly initialized.

These assertions catch issues earlier during development and improve
code reliability.

WhiteBear82 committed:
Improve code naming conventions

The codebase contains naming inconsistencies that do not adhere to the course's prescribed code quality naming standards.

Inconsistent naming can make the code harder to read and maintain, especially in a collaborative learning environment.

Let's make minor adjustments to align variable, function, and class names with the course's naming guidelines. These changes improve readability and maintain consistency across the codebase.

No logic or functionality has been modified as part of these tweaks.

WhiteBear82 committed:
Merge pull request #1 from WhiteBear82/branch-A-CodeQuality

Improve code naming conventions

WhiteBear82 committed:
Refactor to Use Streams

WhiteBear82 committed:
Refactor to Use Streams

WhiteBear82 committed:
Add Help Window

WhiteBear82 committed:
Add Help Window

WhiteBear82 committed:
Add Github CI

WhiteBear82 committed:
Add Github CI

WhiteBear82 committed:
Add error handling

WhiteBear82 committed:
Add error handling

WhiteBear82 committed:
Add more tests

WhiteBear82 committed:
Add more tests

WhiteBear82 committed:
Update user guide

WhiteBear82 committed:
Update user guide

WhiteBear82 committed:
Update user guide

WhiteBear82 committed:
Update user guide

Stickiem commented on src/main/java/wbb/WinterBearBot.java:
```diff
@@ -0,0 +1,126 @@
+                String command = ui.readCommand();
+                assert command != null : "Command input should not be null";
+
+                Command c = parser.parseCommand(command);
```
Should the "c" variable explain more clearly the named entity?

Stickiem commented on src/main/java/wbb/command/AddCommand.java:
```diff
@@ -0,0 +1,71 @@
+     * @throws WBBException if the command cannot be executed correctly.
+     */
+    @Override
+    public void execute(ArrayList<Task> taskList, String command, Ui ui, Storage storage) throws WBBException {
```
Should the "execute()" method contain more details of what it is executing?

Stickiem commented on src/main/java/wbb/util/DateTimeUtility.java:
```diff
@@ -0,0 +1,183 @@
+     * @param day The day in number.
+     * @return The appended format.
+     */
+    public String ordinalDay(int day) {
```
Should the "ordinalDay()" be a verb instead of a noun?

Stickiem commented on src/main/java/wbb/util/DateTimeUtility.java:
```diff
@@ -0,0 +1,183 @@
+     * @param by The deadline.
+     * @return True if the deadline is equals today, otherwise false.
+     */
+    public boolean isDueToday(String by) {
```
Should the "isDueToday()" be a verb like "checkDeadline()", instead of a noun?

Stickiem reviewed (commented):
Looks good.
Some naming changes requested.

WhiteBear82 committed:
Improve Javadoc clarity

WhiteBear82 committed:
Improve Javadoc clarity

WhiteBear82 committed:
Generate release

WhiteBear82 committed:
Generate release

WhiteBear82 committed:
Generate release

WhiteBear82 committed:
Generate release
